src/sentry/snuba/metrics/datasource.py

```diff
@@ -614,6 +619,11 @@ def _fetch_tags_or_values_for_mri(
 
     entity_keys = get_entity_keys_of_use_case_id(use_case_id=use_case_id)
     for entity_key in entity_keys or ():
+        where = []
+        metric_ids = metric_ids_by_entity_key[entity_key]
+        if metric_ids:
```

Comment:
Shouldn't we skip loading the metrics data if, for a specific entity key, we don't have any metric ids?

Reply:
I'm not too sure what this function is used for but it looks like there's a fallback to query all metric ids for some reason. Will see what that's used for first.